brave/browser-laptop
# Add configurable UI scaling

## Break `Preferences > Advanced` into its own file

Auditors: @cezaraugusto, @NejcZdovc

## Test Plan:
1. Open preferences and visit the `Advanced` tab
2. Settings should work as expected; there should be no console errors

--------------

## Add configurable UI scaling
**_NOTE:_** This doesn't affect page size; scaling is only for the browser UI elements

Fixes https://github.com/brave/browser-laptop/issues/1937

Auditors: @bradleyrichter, @alexwykoff, @NejcZdovc

## Test Plan:
1. Open preferences and visit the `Advanced` tab
2. Change the new `Toolbar and UI elements scale` setting.
  - verify default is `Normal`
  - choose a smaller size and verify URL bar, nav buttons, etc get smaller
  - choose a larger size and verify URL bar, nav buttons, etc get larger
  - verify size of page content does not change
3. Open multiple windows and verify changing the setting takes effect
4. Verify setting affects ALL UI elements... such as Alert
5. Verify setting is saved on exit and that all windows are created on launch with the proper size

--------------


## Screenshots

New preference (under advanced)
![screen shot 2017-04-21 at 12 38 50 am](https://cloud.githubusercontent.com/assets/4733304/25267783/fcad8c94-262a-11e7-93ef-8aaecc3138ff.png)

The setting in action
![zoom-ui](https://cloud.githubusercontent.com/assets/4733304/25267847/524f4354-262b-11e7-981c-156116e5e439.gif)
